app/src/main/java/com/sunasterisk/getitdone/ui/home/HomeFragment.kt

```diff
@@ -30,36 +48,128 @@ class HomeFragment : BaseFragment<HomeContract.View, HomePresenter>(), HomeContr
         super.onCreateOptionsMenu(menu, inflater)
     }
 
-    override fun initComponents(savedInstanceState: Bundle?) {
-        presenter.getTaskListFromId(id = listId)
-        presenter.getTasksFromTaskListId(listId = listId)
+    override fun onDestroyView() {
+        super.onDestroyView()
+        presenter?.detach()
     }
 
-    override fun getParentContext() = this.requireContext()
+    override fun initComponents(savedInstanceState: Bundle?) {
+        context?.let {
+            val database = AppDatabase.getInstance(it)
+
+            val taskListRepository = TaskListRepository.getInstance(
+                TaskListLocalDataSource.getInstance(
+                    TaskListDAOImpl.getInstance(database)
+                )
+            )
+
+            val taskRepository = TaskRepository.getInstance(
+                TaskLocalDataSource.getInstance(
+                    TaskDAOImpl.getInstance(database)
+                )
+            )
+
+            presenter = HomePresenter(taskListRepository, taskRepository).also { homePresenter ->
+                homePresenter.attach(this)
+                homePresenter.getTaskListFromId(id = listId)
+                homePresenter.getTasksFromTaskListId(listId = listId)
+            }
+        }
+    }
 
     override fun setToolBarTitle(title: String) {
         toolbarHome.title = title
     }
 
-    override fun displayMessage(message: String) {
+    override fun displayMessage(stringId: Int) {
+        val message = resources.getString(stringId)
         requireContext().toast(message)
```

Comment:
Không dùng requireContext() 

```suggestion
        context?.run {
            toast(getString(stringId))
        }
```